core/src/main/java/tc/oc/pgm/spawner/SpawnerModule.java

```diff
@@ -49,58 +54,86 @@ public SpawnerModule parse(MapFactory factory, Logger logger, Document doc)
             XMLUtils.parseNumber(
-                element.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
+                spawnerEl.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
         Filter playerFilter =
-            filterParser.parseFilterProperty(element, "filter", StaticFilter.ALLOW);
+            filterParser.parseFilterProperty(spawnerEl, "filter", StaticFilter.ALLOW);
 
         List<Spawnable> objects = new ArrayList<>();
-        for (Element spawnable :
-            XMLUtils.getChildren(
-                element, "item")) { // TODO Add more types of spawnables once entity parser is built
-          ItemStack stack = kitParser.parseItem(spawnable, false);
-          SpawnableItem item = new SpawnableItem(stack, numericID);
+        for (Element itemEl : XMLUtils.getChildren(spawnerEl, "item")) {
+          ItemStack stack = kitParser.parseItem(itemEl, false);
+          SpawnableItem item = new SpawnableItem(stack, id);
           objects.add(item);
         }
 
+        for (Element potionEl : XMLUtils.getChildren(spawnerEl, "potion")) {
+          ImmutableList.Builder<PotionEffect> chBuilder = ImmutableList.builder();
+          for (Element potionChild : potionEl.getChildren("effect")) {
+            chBuilder.add(XMLUtils.parsePotionEffect(new InheritingElement(potionChild)));
+          }
+          ImmutableList<PotionEffect> potionChildren = chBuilder.build();
+          if (potionChildren.isEmpty()) {
+            throw new InvalidXMLException("Expected child effects, but found none", spawnerEl);
+          }
+          int damageValue = 0;
+          if (potionEl.getAttribute("damage") != null) {
+            damageValue = XMLUtils.parseNumber(potionEl.getAttribute("damage"), Integer.class, 0);
+          } else {
+            for (PotionEffect potionEffect : potionChildren) {
+              // PotionType lists "true" potions, PotionEffectType "potionEffect.getType()" lists
+              // all possible status effects (ie wither, blindness, etc)
+              // Use the first listed PotionType for potion color
+              if (PotionType.getByEffect(potionEffect.getType()) != null) {
+                damageValue = PotionType.getByEffect(potionEffect.getType()).getDamageValue();
+                break;
+              }
+            }
+          }
+          objects.add(new SpawnablePotion(potionChildren, damageValue, id));
+        }
+
         SpawnerDefinition spawnerDefinition =
             new SpawnerDefinition(
+                id,
                 objects,
                 spawnRegion,
                 playerRegion,
                 playerFilter,
                 delay,
                 minDelay,
                 maxDelay,
-                maxEntities,
-                numericID);
-        factory.getFeatures().addFeature(element, spawnerDefinition);
+                maxEntities);
+        factory.getFeatures().addFeature(spawnerEl, spawnerDefinition);
```

Comment:
i *think* you may no longer need the EL here, since the definition is self identifying now, but double-check (try to just pass spawnerDefinition)

Reply:
The element is still required.